web/src/modules/layout/components/HeaderActionItemDropDown.js

```diff
@@ -64,63 +62,45 @@ export const DropDownContainer = styled.div`
 `
 
 type PropsType = {|
-  children: React.Element<*>,
+  children: (closeDropDown: () => void) => React.Element<*>,
   theme: ThemeType,
   iconSrc: string,
   text: string
 |}
 
-type StateType = {|
-  dropDownActive: boolean
-|}
-
 /**
  * Designed to work as an item of a HeaderActionBar. Once clicked, the child node becomes visible right underneath the
  * Header. Once the user clicks outside, the node is hidden again. Additionally, the inner node gets a
  * closeDropDownCallback through its props to close the dropDown and hide itself.
  */
-export class HeaderActionItemDropDown extends React.Component<PropsType, StateType> {
-  componentDidUpdate () {
-    /* https://www.npmjs.com/package/react-tooltip#1-using-tooltip-within-the-modal-eg-react-modal- */
-    ReactTooltip.rebuild()
-  }
+const HeaderActionItemDropDown = (props: PropsType) => {
+  const { iconSrc, text, children, theme } = props
+  const [dropDownActive, setDropDownActive] = useState(false)
 
-  constructor (props: PropsType) {
-    super(props)
-    this.state = { dropDownActive: false }
-  }
+  const toggleDropDown = useCallback(() => {
+    setDropDownActive(!dropDownActive)
+  }, [setDropDownActive, dropDownActive])
 
-  toggleDropDown = () => {
-    this.setState(prevState => ({ dropDownActive: !prevState.dropDownActive }))
+  const closeDropDown = () => {
+    setDropDownActive(false)
   }
 
-  closeDropDown = () => {
-    if (this.state.dropDownActive) {
-      this.toggleDropDown()
-    }
+  const clickOutside = () => {
+    setDropDownActive(false)
   }
 
-  handleClickOutside = () => {
-    this.closeDropDown()
-  }
-
-  render () {
-    const { iconSrc, text, children, theme } = this.props
-    const { dropDownActive } = this.state
-
-    return (
-      <Container theme={theme}>
-        <button selector='button' data-tip={text} data-event='mouseover' data-event-off='click mouseout' aria-label={text} onClick={this.toggleDropDown}>
-          <img alt='' src={iconSrc} />
-        </button>
-        <DropDownContainer active={dropDownActive} theme={theme}>
-          {React.cloneElement(children, {
-            closeDropDownCallback: this.closeDropDown
-          })}
-        </DropDownContainer>
-      </Container>
-    )
-  }
+  const wrapperRef = useRef(null)
+  useOnClickOutside(wrapperRef, clickOutside)
+
+  return <Container ref={wrapperRef} theme={theme}>
+    <button selector='button' data-tip={text} data-event='mouseover' data-event-off='click mouseout' aria-label={text}
+            onClick={toggleDropDown}>
+      <img alt='' src={iconSrc} />
+    </button>
+    {dropDownActive && <DropDownContainer active={dropDownActive} theme={theme}>
```

Comment:
I think the active logic of the `DropDownContainer` is not required anymore as it is always active.

Reply:
Shouldn't we also use `xy ? <asdf /> : null` instead of `xy && <asdf />`?

Reply:
Not sure did this cause problems?

Reply:
Not sure, I think it's more about typing.